### PR TITLE
Badges coremod improvements

### PR DESCRIPTION
**Pull request by asportnoy**
- Create new `gotoOrJoinServer` util
- Add click events for more replugged badges, including joining the server
- Use common `Tooltip` component

### CODE REVIEW

File: src/renderer/coremods/badges/badge.tsx [hunk 1]
```diff
@@ -35,21 +24,25 @@ interface BadgeProps {
...
+export const Base = ({
+  color,
+  tooltip,
+  tooltipPosition,
+  className,
+  children,
+  gap,
+  onClick,
+}: BadgeProps): React.ReactElement => {
   // todo: common component
```
Comment (12944qwerty) [inline src/renderer/coremods/badges/badge.tsx:36]: Just make it a common component rn lol

File: src/renderer/coremods/badges/badge.tsx [hunk 2]
```diff
@@ -71,108 +64,137 @@ export const getBadgeComponent = async (): Promise<(args: BadgeProps) => React.R
...
-      <Clickable className={clickableClass.clickable} onClick={onClick || (() => undefined)}>
+  const child = (
+    <div
+      className={`${profileBadge22} replugged-badge ${className || ""}`}
+      style={{ color: `#${color || "7289da"}` }}>
+      {children}
+    </div>
+  );
+  return (
+    <Clickable className={clickableClass.clickable} onClick={onClick || (() => undefined)}>
```
Comment (12944qwerty) [inline src/renderer/coremods/badges/badge.tsx:75]: If onClick isn't defined, then the badge shouldn't be clickable

File: src/renderer/coremods/badges/badge.tsx [hunk 3]
```diff
@@ -71,108 +64,137 @@ export const getBadgeComponent = async (): Promise<(args: BadgeProps) => React.R
...
-            </span>
-          )}
+          {child}
         </Tooltip>
-      </Clickable>
-    );
-  };
+      ) : (
+        child
+      )}
```
Comment (12944qwerty) [inline src/renderer/coremods/badges/badge.tsx:85]: There is a `shouldShow` prop that does this for you lol

```jsx
<Tooltip
  shouldShow={Boolean(tooltip)}
  text={tooltip!}
  position={tooltipPosition || "top"}
  spacing={gap === false ? 0 : 12}>
  {child}
</Tooltip>
```

**Review by 12944qwerty — CHANGES_REQUESTED** [review]

File: src/renderer/util.ts [hunk 4]
```diff
@@ -75,3 +78,87 @@ export function forceUpdateElement(selector: string, all = false): void {
 
   elements.forEach((element) => getOwnerInstance(element)?.forceUpdate());
 }
+
+type Invite = Record<string, unknown> & {
```
[the remark this answers is not in the document]
Comment (asportnoy) [reply src/renderer/util.ts:82]: It's not in `discord-types`. This is only a partial type for this one specific usage, so unless there's use for it elsewhere, I think we should leave it here.

File: src/renderer/coremods/badges/badge.tsx (hunk 2 above)
Comment (asportnoy) [reply src/renderer/coremods/badges/badge.tsx]: The clickable class gives it the cursor, which I think it should have due to the tooltip.

File: src/renderer/coremods/badges/badge.tsx (hunk 3 above)
Comment (asportnoy) [reply src/renderer/coremods/badges/badge.tsx:65]: I think the current way of doing it is better. No reason to include extra elements if they're not being used.

Comment (12944qwerty) [reply src/renderer/coremods/badges/badge.tsx:65]: It's not an extra element... it's just a prop that's already added. You';d just have to replace those lines with the codeblock I sent

Comment (asportnoy) [reply src/renderer/coremods/badges/badge.tsx:65]: If the element isn't doing anything then it is an extra element.

File: src/renderer/coremods/badges/badge.tsx (hunk 2 above)
Comment (12944qwerty) [reply src/renderer/coremods/badges/badge.tsx]: I meant that if onclick isn't defined, then the cursor shouldn't be the clickable type of cursor lol. If it doesn't do anything, there shouldn't be any need to have the clickable cursor.

File: src/renderer/util.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (12944qwerty) [reply src/renderer/util.ts:82]: It's still a discord type.

File: src/renderer/coremods/badges/badge.tsx (hunk 2 above)
Comment (asportnoy) [reply src/renderer/coremods/badges/badge.tsx]: All of the Discord badges show a pointer even if they don't do anything onclick.

File: src/renderer/coremods/badges/badge.tsx (hunk 3 above)
Comment (12944qwerty) [reply src/renderer/coremods/badges/badge.tsx:65]: You're literally reducing lines into one prop line?

**Review by 12944qwerty — APPROVED** [review]
for a sec I was like "to test join server, I need to leave and then only I can join... hmmm"

**Pull request merged**